src/Core/Module/ModuleOverrideVerifier.php

```diff
@@ -0,0 +1,129 @@
+    {
+        // if module doesn't have an override folder, there is no conflict
+        if (!is_dir($moduleOverridePath)) {
+            return false;
+        }
+
+        $tools = new Tools();
+        // module has overrides, let's check override files one by one
+        foreach ($tools->scandir($moduleOverridePath, 'php', '', true) as $file) {
+            $module_override_path = $moduleOverridePath . '/' . $file;
+            $prestashop_override_path = _PS_OVERRIDE_DIR_ . $file;
+
+            if (file_exists($prestashop_override_path)) {
+                if ($this->hasConflictingMethod($module_override_path, $prestashop_override_path)) {
+                    $this->errors[] = $this->translator->trans(
+                        'The override file %1$s conflicts with an existing override in %2$s.',
+                        [$module_override_path, $prestashop_override_path],
+                        'Admin.Modules.Notification'
+                    );
+                }
+            }
+        }
+
+        return !empty($this->errors);
+    }
+
+    public function getErrors(): array
+    {
+        return $this->errors;
+    }
+
+    /*
+     * Checks if a module override class has a method that is already overridden by another module
+     */
+    private function hasConflictingMethod(string $module_override_path, string $prestashop_override_path): bool
+    {
+        $module_methods = $this->getClassMethodsFromContent(file_get_contents($module_override_path));
+        $prestashop_methods = $this->getClassMethodsFromContent(file_get_contents($prestashop_override_path));
+
+        foreach ($module_methods as $method) {
+            if (in_array($method, $prestashop_methods)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /*
+     * This function parses php file content and gets a list of methods from its content.
+     */
+    private function getClassMethodsFromContent(string $content): array
+    {
+        // get list of tokens, see https://www.php.net/manual/en/function.token-get-all.php
+        $tokens = token_get_all($content);
+        $methods = [];
+        foreach ($tokens as $index => $token) {
+            $method_name = null;
+            // filter method definitions
+            if (is_array($token) && $token[0] === T_FUNCTION) {
+                // this loop is necessary in case some additional spaces exist before function name
+                for ($i = $index + 1; $i < count($tokens); ++$i) {
+                    if (is_array($tokens[$i]) && $tokens[$i][0] === T_STRING) {
+                        $method_name = $tokens[$i][1];
+                        break;
+                    }
+                }
+
+                // collect method name
+                if ($method_name) {
+                    $methods[] = $method_name; // Save method name
+                }
+            }
+        }
+
+        return $methods;
+    }
```

Comment:
```suggestion
    private function getClassMethodsFromContent(string $content): array
    {
        // Get the list of tokens, see https://www.php.net/manual/en/function.token-get-all.php
        $tokens = token_get_all($content);
        $methods = [];
        foreach ($tokens as $index => $token) {
            $methodName = null;
            // filter method definitions
            if (is_array($token) && $token[0] === T_FUNCTION) {
                // This loop is necessary in case some additional spaces exist before the function name
                for ($i = $index + 1; $i < count($tokens); ++$i) {
                    if (is_array($tokens[$i]) && $tokens[$i][0] === T_STRING) {
                        $methodName = $tokens[$i][1];
                        break;
                    }
                }

                // Collect method name
                if ($methodName) {
                    $methods[] = $methodName; // Save method name
                }
            }
        }

        return $methods;
    }
```